Updates to SCIM docs (#7200)
dbt-labs/docs.getdbt.com: website/docs/docs/cloud/manage-access/scim.md

sidebar: "Set up SCIM"
---

# Set up SCIM <Lifecycle status="beta, enterprise" />

The System for Cross-Domain Identity Management (SCIM) makes user data more secure and simplifies the admin and end-user lifecycle experience by automating user identities and groups. You can create or disable user identities in your Identity Provider (IdP), and SCIM will automatically make those changes in near real-time downstream in dbt Cloud.

## Prerequisites 

To configure SCIM in your dbt Cloud environment:
- You must be on an Enterprise plan.
- You must be using Okta as your SSO provider.

> - You must have proper permissions to configure the account settings in [dbt Cloud](/docs/cloud/manage-access/enterprise-permissions) and alter app settings in [Okta](https://help.okta.com/en-us/content/topics/security/administrators-admin-comparison.htm).

Reviewer: ```suggestion
- You must have permissions to configure the account settings in [dbt Cloud](/docs/cloud/manage-access/enterprise-permissions) and change application settings in [Okta](https://help.okta.com/en-us/content/topics/security/administrators-admin-comparison.htm).
```

> - If you have IP restrictions enabled, you must add Okta's IPs to your allowlist.

Reviewer: ```suggestion
- If you have IP restrictions enabled, you must add [Okta's IPs](https://help.okta.com/en-us/content/topics/security/ip-address-allow-listing.htm) to your allowlist.
```

### Supported features

The currently available supported features for SCIM are:

- User provisioning and de-provisioning
- User profile updates
- Group creation and management
- Importing groups and users

When users are provisioned, the following attributes are supported

> - Username

Reviewer: do we want to use the user attributes instead? e.g. `userName`, `givenName`, `familyName`? or is that for the api?

Please complete the [setup SSO with Okta](/docs/cloud/manage-access/set-up-sso-okta) steps before configuring SCIM settings.

### Set up dbt Cloud

To retrieve the necessary dbt Cloud configurations for use in Okta:

1. Navigate to your dbt Cloud **Account settings**.
2. Select **Single sign-on** from the left-side menu.
3. Scroll to the bottom of your Okta configuration settings and click **Enable SCIM**.

>     <Lightbox src="/img/docs/dbt-cloud/access-control/enable-scim.png" width="60%" title="SCIM enabled in the Okta configuration settings." />

Reviewer: just double checking if this is a real account number in the screenshot?

Author: Yes but no concerns with exposing. Wanted unredacted screenshot for the workflow

4. Record the **SCIM base URL** field for use in a later step.
5. Click **Create SCIM token**.
6. In the pop-out window, give the token a name that will make it easily identifiable. Click **Save**.
    <Lightbox src="/img/docs/dbt-cloud/access-control/name-scim-token.png" width="60%" title="Give your token and identifier." />
7. Copy the token and record it securely, as _it will not be available again after you close the window_. You must create a new token if you lose the current one. 
    <Lightbox src="/img/docs/dbt-cloud/access-control/copy-scim-token.png" width="60%" title="Give your token and identifier." />

> 8. (Optional) Manual updates are turned off by default for all SCIM-managed entities, including the ability to invite new users manually. This ensures SCIM-managed entities stay in sync with the IdP, and we recommend keeping this setting disabled.

Reviewer: im pretty sure this version is right but posting it just in case: based on the screenshot below, it sounds like we want to keep it disabled to allow the sync. but this [notion doc](https://www.notion.so/dbtlabs/SCIM-Beta-Documentation-173bb38ebda78046a9bfeb7a8ff40c74?pvs=4#1babb38ebda780e3b96df332459b992f) says to keep it enabled so it stays in insync?

Author: It's worded strangely in the notion doc for sure lol. This text here is what it should say

>    - However, if you need to make manual updates (for example, if you want to update group membership for a SCIM-managed group), you can enable this setting by clicking **Allow manual updates**.

Reviewer: ```suggestion
   - However, if you need to make manual updates (like update group membership for a SCIM-managed group), you can enable this setting by clicking **Allow manual updates**.
```

>     <Lightbox src="/img/docs/dbt-cloud/access-control/scim-manual-updates.png" width="60%" title="Enabling manual updates in SCIM settings." />

Reviewer: it was a little too small to read
```suggestion
    <Lightbox src="/img/docs/dbt-cloud/access-control/scim-manual-updates.png" width="70%" title="Enabling manual updates in SCIM settings." />
```

:::note License mapping

> SCIM groups are mapped to dbt Cloud groups, so license mappings can be used to assign licenses to SCIM groups using the group name as an identifier. Currently, setting an attribute on the SCIM group for license type is not supported. 

Reviewer: suggesting active voice here
```suggestion
dbt Cloud maps SCIM groups to its own groups, so you can assign licenses to SCIM groups using the group name as an identifier. Currently, setting a license type directly as an attribute on the SCIM group isn't supported.
```

:::

### Set up Okta

1. Log in to your Okta account and locate the app configured for the dbt Cloud SSO integration.
2. Navigate to the **General** tab and ensure **Enable SCIM provisioning** is checked or the **Provisioning** tab will not be displayed. 
    <Lightbox src="/img/docs/dbt-cloud/access-control/scim-provisioned.png" width="60%" title="Enable SCIM provisioning in Okta." />
3. Open the **Provisioning** tab and select **Integration**.

> 4. Paste the **SCIM base URL** from dbt Cloud to the first field and enter your preferred **Unique identifier field for users** (`userName` is recommended) next.

Reviewer: might be helpful to link to the earlier section that mentions this
```suggestion
4. Paste the [**SCIM base URL** from dbt Cloud](#set-up-dbt-cloud) to the first field, then enter your preferred **Unique identifier field for users** &mdash; we recommend `userName`.
```

5. Click the checkboxes for the following **Supported provisioning actions**:
    - Push New Users
    - Push Profile Updates
    - Push Groups
6. From the **Authentication mode** dropdown, select **HTTP Header**.
7. In the **Authorization** section, paste the token from dbt Cloud into the **Bearer** field.
    <Lightbox src="/img/docs/dbt-cloud/access-control/scim-okta-config.png" width="60%" title="The completed SCIM configuration in the Okta app." />
8. Ensure that the following provisioning actions are selected:
    - Create users
    - Update user attributes
    - Deactivate users
    <Lightbox src="/img/docs/dbt-cloud/access-control/provisioning-actions.png" width="60%" title="Ensure the users are properly provisioned with these settings." />

9. Test the connection and click **Save** once completed. 

You've now configured SCIM for the Okta SSO integration in dbt Cloud.

> ### Existing integrations

Author: ```suggestion
### Existing Okta integrations
```

If you are adding SCIM to an existing Okta integration in dbt Cloud (as opposed to setting up SCIM and Oauth concurrently for the first time), there is some functionality you should be aware of:

> - Users and groups already synced to dbt Cloud will become SCIM-managed once the SCIM configuration is completed.

Reviewer: ```suggestion
- Users and groups already synced to dbt Cloud will become SCIM-managed once you complete the SCIM configuration. You won't need to create duplicate users when setting up SCIM.
```

Author: That sentence is more to put at ease than a functional part. I recommend leaving it out to avoid creating unease that doesn't need to exist.
```suggestion
- Users and groups already synced to dbt Cloud will become SCIM-managed once you complete the SCIM configuration.
```

- You can leverage SCIM to import and manage existing dbt Cloud groups. Update the groups in your IdP with the same naming convention used for dbt Cloud groups. New users, groups, and existing profile changes will be automatically imported into dbt Cloud.

>     - Ensure the **Import users and groups** checkbox is selected in the **Provisioning settings** tab in the Okta SCIM configuration.

Author: ```suggestion
    - Ensure the **Import users and profile updates** and **Import groups** checkboxes are selected in the **Provisioning settings** tab in the Okta SCIM configuration.
```

>     - Ensure the **Import users and groups** checkbox is selected in the **Provisioning settings** tab in the Okta SCIM configuration.

Reviewer: is it import users and groups or import new users and profile updates? not sure if the screenshot in the notion doc is the right one i should be referencing?

Author: There are apparently two different okta experiences. The screenshot in the docs is from the less common one but I'll have them doublecheck the language
